packages/batch-submitter/src/batch-submitter/state-batch-submitter.ts

```diff
@@ -177,6 +177,8 @@ export class StateBatchSubmitter extends BatchSubmitter {
       batch,
       offsetStartsAtIndex
     )
```

Comment:
It seems more canonical ethers style to do so:

```
    const tx = await this.chainContract.populateTransaction.appendStateBatch(
      batch,
      offsetStartsAtIndex,
    { nonce: await this.signer.getTransactionCount }
    )
```

Reply:
My reasoning behind this is that malleating the transaction is generally a bad idea - in this case its safe because `populateTransaction` does not sign the transaction, but a future refactor or copypasta could result in unsafe code